### DISCUSSION TITLE
Depending on `rich` package for output formatting

### DISCUSSION THREAD

**Discussion by YannickJadoul** [pypa/cibuildwheel#525] [Ideas]
@henryiii just mentioned https://github.com/willmcgugan/rich (in the context of pybind11).

This seems like a nice and powerful project which would be a minor dependency (182.5 kB wheels: https://pypi.org/project/rich/#files), so it might be nice to use this in our `logger.py` output (and e.g. replace the manual `Colors` class by something externally maintained).

**Comment by henryiii** [comment]
I've played with it for a day, and it is insanely impressive (look through the beautiful readme); I've written a color library before (plumbum.colors), so think I am allowed to be impressed. :) The custom tracebacks (with locals!), and the indented color-syntax highlighted logging is just nuts. Also the protocol for custom formatting for user objects. Only having played with it for a day, I have no idea about how well does hold up in CI logs; though like plumbum.colors, you can control the console settings. Someone at work (as in Princeton's RSE group) uses it, and his only complaint is that it doesn't work well in PyCharm (known issue).
